Source/WebCore/animation/CSSPropertyAnimation.cpp

```diff
@@ -1079,6 +1079,79 @@ class LengthVariantPropertyWrapper final : public PropertyWrapperGetter<const T&
     void (RenderStyle::*m_setter)(T&&);
 };
 
+class OptionalLengthPropertyWrapper : public PropertyWrapperGetter<std::optional<Length>> {
+    WTF_MAKE_FAST_ALLOCATED;
+
+public:
+    enum class Flags {
+        IsLengthPercentage = 1 << 0,
+        NegativeLengthsAreInvalid = 1 << 1,
+    };
+    OptionalLengthPropertyWrapper(CSSPropertyID property, std::optional<Length> (RenderStyle::*getter)() const, void (RenderStyle::*setter)(std::optional<Length>), OptionSet<Flags> flags = { })
+        : PropertyWrapperGetter<std::optional<Length>>(property, getter)
+        , m_setter(setter)
+        , m_flags(flags)
+    {
+    }
+
+protected:
+    bool canInterpolate(const RenderStyle& from, const RenderStyle& to, CompositeOperation) const
+    {
+        if (!this->value(from).has_value() || !this->value(to).has_value())
+            return false;
+
+        bool isLengthPercentage = m_flags.contains(Flags::IsLengthPercentage);
+        return canInterpolateLengths(this->value(from).value(), this->value(to).value(), isLengthPercentage);
+    }
+
+    void blend(RenderStyle& destination, const RenderStyle& from, const RenderStyle& to, const CSSPropertyBlendingContext& context) const
+    {
+        if (context.isDiscrete) {
+            (destination.*m_setter)(context.progress < 0.5 ? this->value(from) : this->value(to));
```

Comment:
You don't need to check for the `0.5` boundary here since you're guaranteed to get a `0` or `1` value since this is a discrete interpolation. So this can be simplified to:

```
(destination.*m_setter)(this->value(context.progress ? to : from))
```

You can also add this:

```
ASSERT(!context.progress || context.progress == 1);
```

Reply:
Done, thanks!